### PR TITLE
New JWT interface and updated libraries

### PR DESCRIPTION
**Pull request by toby**
* Replace outdated github.com/dgrijalva/jwt-go library with github.com/golang-jwt/jwt/v4
* Update github.com/auth0/go-jwt-middleware to v2
* Check issuer and audience claims fields
* Use RegisteredClaims instead of StandardClaims
* Allow for custom audience issuing in the API and CLI with `charm jwt [aud...]`

### CODE REVIEW

**Comment by toby** [comment]
As per the note [here](https://stackoverflow.com/a/66894047), https://jwt.io doesn't support Ed25519 validation, so it claims the signature is invalid. It seems to be working fine though.

File: cmd/serve.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var (
 				cfg.DataDir = serverDataDir
 			}
 			sp := fmt.Sprintf("%s/.ssh", cfg.DataDir)
-			kp, err := keygen.NewWithWrite(sp, "charm_server", []byte(""), keygen.RSA)
+			kp, err := keygen.NewWithWrite(sp, "charm_server", []byte(""), keygen.Ed25519)
```
Comment (caarlos0) [inline cmd/serve.go:36]: wont this trigger SSH key check errors on existing servers?

e.g.: there was a server using a RSA key, and now its a ED25519 key, which is different from the previous one...

File: server/ssh.go [hunk 2]
```diff
@@ -84,6 +78,46 @@ func (me *SSHServer) authHandler(ctx ssh.Context, key ssh.PublicKey) bool {
...
+	j, err := me.newJWT(u.CharmID, aud)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	_, _ = s.Write([]byte(j))
+	me.config.Stats.JWT()
+}
+
+func (me *SSHServer) newJWT(charmID string, audience []string) (string, error) {
```
Comment (caarlos0) [inline server/ssh.go:109]: audience could be a `...string` to avoid typing `[]string{"item"}` 💅🏻

File: cmd/serve.go (hunk 1 above)
Comment (toby) [reply cmd/serve.go:36]: There's no host key verification in `charm` so it _should_ be ok.

**Review by aymanbagabas — APPROVED** [review]

File: server/ssh.go (hunk 2 above)
Comment (toby) [reply server/ssh.go]: Good call, just fixed this.

**Pull request merged**